src/SqlConverters.cs

```diff
@@ -186,7 +186,12 @@ public virtual async Task<string> BuildItemFromAttributeAsync(SqlAttribute attri
                     var dataTable = new DataTable();
                     adapter.Fill(dataTable);
                     this._logger.LogInformation($"{dataTable.Rows.Count} row(s) queried from database: {connection.Database} using Command: {command.CommandText}");
-                    return JsonConvert.SerializeObject(dataTable);
+                    // Serialize any DateTime objects in UTC format
+                    var jsonSerializerSettings = new JsonSerializerSettings()
+                    {
+                        DateFormatString = "yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'fffZ"
```

Comment:
So we do a slightly different format in the AsyncCollector - https://github.com/Azure/azure-functions-sql-extension/blob/b966b42f3d74d872d5c5b63b40f35daf97669b8c/src/SqlAsyncCollector.cs#L380

But it looks like both that and with the Z are supported formats : 

```
The ISO 8601 date with time format is as follows:

YYYY-MM-DDThh:mm:ss[.nnnnnnn][{+|-}hh:mm]

YYYY-MM-DDThh:mm:ss[.nnnnnnn]Z (UTC, Coordinated Universal Time)
```

We definitely need to be consistent across the entire codebase with how we treat datetimes so I would suggest pulling that constant out of that class and putting it in a general constants file, then using that both there and here. 

In addition, please make a follow up issue to investigate all serialization/deserialization to ensure we're doing it correctly (including in the triggerbinding branch).

Reply:
I will create a constants file and include Z in the format since gson expects the time zone. 
Created an issue to track here: https://github.com/Azure/azure-functions-sql-extension/issues/413